sdk/core/core/src/az_json_parser.c

```diff
@@ -73,7 +63,7 @@ az_json_parser_stack_last(az_json_parser const* json_parser)
 AZ_NODISCARD AZ_INLINE az_result
 az_json_parser_push_stack(az_json_parser* json_parser, _az_json_stack json_stack)
 {
-  if (json_parser->_internal.stack >> AZ_JSON_STACK_SIZE != 0)
+  if (json_parser->_internal.stack >> (_az_MAX_JSON_STACK_SIZE - 1) != 0)
```

Comment:
Add comments to this code.  Its not obvious how we use stack and why we are shifting to check for 'full'

Reply:
I intentionally tried to keep the `az_json_parser` changes to a minimum because these things will likely change in the future (to leverage the common stack implementation) which is why I didn't comment this (focusing primarily on the builder). I would like to defer this until I fully understand the parser implementation.